tools/configen/configen/config.py

```diff
@@ -10,6 +10,7 @@
 class ModuleConf:
     name: str = MISSING
     classes: List[str] = MISSING
+    default_flags: Optional[Dict[str, Any]] = field(default_factory=dict)
```

Comment:
Here is a thought, instead of using a dict and then restricting manually at runtime to supported types, we can just use a dataclass directly.
dunno what I was thinking.
```python
@dataclass
class Flags:
  _recursive_ : Optional[bool] =  None
  _convert_ : Optional[str] =  None
```

For convert, you should actually use the ConvertMode Enum from Hydra directly here.

```python
  _convert_ : Optional[ConvertMode] =  None
```